### ISSUE TITLE
[Cloud Docker] env.php configuration is lost during a deploy

### ISSUE THREAD

**Issue by joeshelton-wagento** [magento/ece-tools#511]
With Docker in developer mode, env.php is regenerated during a deploy. Configurations from before the deploy are lost. Also, the array format changes from `[]` to `array()`.

### Preconditions
Working Docker environment with the following in env.php. Notice the disabling of Google Analytics.

```php
...

    'install' => [
        'date' => 'Thu, 06 Oct 2016 09:48:22 +0000'
    ],
    'system' => [
        'default' => [
            'catalog' => [
                'search' => [
                    'engine' => 'elasticsearch5',
                    'elasticsearch5_server_hostname' => 'elasticsearch',
                    'elasticsearch5_server_port' => '9200'
                ]
            ],
            'google' => [
                'analytics' => [
                    'active' => '0'
                ]
            ]
        ]
    ],
    'cache' => [
        'frontend' => [
            'default' => [
                'backend' => 'Cm_Cache_Backend_Redis',
                'backend_options' => [
                    'server' => 'redis',
                    'port' => '6379',
                    'database' => 1
                ]
            ],
            'page_cache' => [
                'backend' => 'Cm_Cache_Backend_Redis',
                'backend_options' => [
                    'server' => 'redis',
                    'port' => '6379',
                    'database' => 2
                ]
            ]
        ]
    ],

...
```

### Steps to reproduce
`docker-compose run deploy cloud-deploy`

### Expected result
GA would continue to be disabled in env.php.

### Actual result
GA configuration was lost.

```php
...

  'install' => 
  array (
    'date' => 'Thu, 06 Oct 2016 09:48:22 +0000',
  ),
  'system' => 
  array (
    'default' => 
    array (
      'catalog' => 
      array (
        'search' => 
        array (
          'engine' => 'elasticsearch5',
          'elasticsearch5_server_hostname' => 'elasticsearch',
          'elasticsearch5_server_port' => '9200',
        ),
      ),
    ),
  ),
  'cache' => 
  array (
    'frontend' => 
    array (
      'default' => 
      array (
        'backend' => 'Cm_Cache_Backend_Redis',
        'backend_options' => 
        array (
          'server' => 'redis',
          'port' => '6379',
          'database' => 1,
        ),
      ),
      'page_cache' => 
      array (
        'backend' => 'Cm_Cache_Backend_Redis',
        'backend_options' => 
        array (
          'server' => 'redis',
          'port' => '6379',
          'database' => 2,
        ),
      ),
    ),
  ),

...
```

**Comment by YPyltiai** [comment]
Hi @joeshelton-wagento,

Thanks for opening the issue.

Looks like a bug for me. It works this way be design, since ECE-Tools generates env.php on deploy, but for dev mode it doesn't make sense for sure. We'll need to think how to change it without modifying actual production deploy phase.

**Comment by joeshelton-wagento** [comment]
In the Cloud environment, env.php is regenerated with every deployment. But, custom configuration changes are preserved, as long as they are not specific to a service. Expecting the same behavior from the Docker environment.

**Comment by YPyltiai** [comment]
I see. It explains the issue now. @andriyShevtsov / @mveeramneni , please validate the steps to reproduce.

Thanks!

**Comment by andriyShevtsov** [comment]
Hello @joeshelton-wagento. Thank you. I created ticket MAGECLOUD-3668

**Comment by YPyltiai** [comment]
Identified as being caused by #518 
@joeshelton-wagento , it looks like a regression, not an actual Docker bug.

**Comment by YPyltiai** [comment]
Fixed in #523
